axelrod/strategies/titfortat.py

```diff
@@ -91,7 +91,36 @@ class TwoTitsForTat(Player):
     @staticmethod
     def strategy(opponent: Player) -> Action:
         return D if D in opponent.history[-2:] else C
+    
+class DynamicTitForTat(Player):
+    """A player starts by cooperating and then mimics previous move by
+    opponent with a dynamic bias based off of the opponents ratio of
+    defections to cooperations towards cooporating regardless of 
+    the move."""
+    
+    name = 'Dynamic Tit for Tat'
+    classifier = {
+        'memory_depth': 2,  # Long memory, memory-2
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
 
+    @staticmethod
+    def strategy(self, opponent):
+        try:
```

Comment:
Instead of using `try` `except` clause we usually just check the len of `opponent.history` so:

```python
if len(opponent.history) == 0:
    return C
...
```

However in this case, as you're using `opponent.history[-2:]`  I believe you don't need a check anyway:

```python
>>> history = []
>>> history[-2:]
[]
>>> history = ["C"]
>>> history[-2:]
["C"]
```

So you don't need this `try` `except` clause at all (I believe).